src/main/java/appeng/items/parts/ItemMultipartRendering.java

```diff
@@ -0,0 +1,33 @@
+package appeng.items.parts;
+
+
+import net.minecraft.client.renderer.block.model.ModelResourceLocation;
+import net.minecraft.item.ItemStack;
+
+import appeng.bootstrap.IItemRendering;
+import appeng.bootstrap.ItemRenderingCustomizer;
+
+
+public class ItemMultipartRendering extends ItemRenderingCustomizer
+{
+
+	private final ItemMultiPart item;
+
+	public ItemMultipartRendering( ItemMultiPart item )
+	{
+		this.item = item;
+	}
+
+	@Override
+	public void customize( IItemRendering rendering )
+	{
+		rendering.meshDefinition( this::getItemMeshDefinition );
+	}
+
+	private ModelResourceLocation getItemMeshDefinition( ItemStack is )
+	{
+		// TODO: Avoid object creation here. This happens every frame
```

Comment:
Even when leaving the scope of the method. Or also replacing objects with primitives as parameters. Sometimes there is way to much magic involved.

If trivial, there is nothing against replacing it. But if we run into cache invalidation issues because it has to map the `ItemStack` dynamically to a `ModelResourceLocation` based on nbt data, it is probably an idea to leave it until there is actually a problem.